### PR TITLE
Sscs 4030 questions into pdf

### PR DESCRIPTION
**Pull request by chrisdavecm**

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @Test
+    public void testCreate() throws IOException, InterruptedException {
+        OnlineHearing onlineHearing = createHearingWithQuestion(true);
+
+        answerQuestion(onlineHearing.getHearingId(), onlineHearing.getQuestionId());
+
+        //now trigger our endpoint
+        resolveHearing(onlineHearing.getHearingId(), onlineHearing.getCaseId());
+    }
+
```
Comment (chrisgrimble) [inline src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java:19]: Test true is true!

File: src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+public class CreateHearingPdfTest extends BaseFunctionTest {
+
+    @Test
+    public void testCreate() throws IOException, InterruptedException {
+        OnlineHearing onlineHearing = createHearingWithQuestion(true);
+
+        answerQuestion(onlineHearing.getHearingId(), onlineHearing.getQuestionId());
+
+        //now trigger our endpoint
+        resolveHearing(onlineHearing.getHearingId(), onlineHearing.getCaseId());
```
Comment (chrisgrimble) [inline src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java:17]: Is there something we can test for here to check we have resolved the hearing.

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/Application.java [hunk 3]
```diff
@@ -52,4 +57,26 @@ public AuthTokenGenerator serviceAuthTokenGenerator(
     public RestTemplate restTemplate() {
         return new RestTemplate();
     }
+
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/sscscorbackend/Application.java:60]: Fields at the top of the object.

Are these all bits to be deleted when we split out the common library and we don't really need? Is so keep it all together.

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/OnlineHearingPdfWraper.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.reform.sscscorbackend.domain.onlinehearing;
+
+import lombok.Builder;
+import lombok.Value;
+import uk.gov.hmcts.reform.sscscorbackend.domain.CohQuestionRounds;
+
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/OnlineHearingPdfWraper.java:7]: We haven't used Lombok in any other places in COR.

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/CohEvent.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.sscscorbackend.domain.onlinehearing;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Builder;
+
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/CohEvent.java:5]: We haven't used Lombok in any other places in COR.

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java [hunk 6]
```diff
@@ -20,22 +23,35 @@
 import uk.gov.hmcts.reform.sscscorbackend.domain.Evidence;
 import uk.gov.hmcts.reform.sscscorbackend.service.documentmanagement.DocumentManagementService;
 
+@Slf4j
 @Service
 public class EvidenceUploadService {
     private final DocumentManagementService documentManagementService;
     private final CcdService ccdService;
     private final IdamService idamService;
-    private final OnlineHearingService onlineHearingService;
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java:28]: Don't really understand what the issue with this being final and then passed in via the constructor. Think we have only used the setter straight after the constructor.

Comment (chrisgrimble) [reply src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java:28]: Ah we have a circular reference we should probably fix that.

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/OnlineHearingService.java [hunk 7]
```diff
@@ -1,35 +1,62 @@
 package uk.gov.hmcts.reform.sscscorbackend.service;
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/OnlineHearingService.java:1]: Same comments as EvidenceUploadService maybe is we split this into two as PDF and OnlineHearingService we could break the circular dependency.

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java [hunk 8]
```diff
@@ -99,7 +115,28 @@ private void addDocumentToCcd(String questionId, Long ccdCaseId, Document docume
...
 
-        ccdService.updateCase(caseDetails.getData(), ccdCaseId, "uploadCorDocument", "SSCS - cor evidence uploaded", "Updated SSCS", idamTokens);
+        ccdService.updateCase(caseDetails.getData(), ccdCaseId, UPLOAD_COR_DOCUMENT, "SSCS - cor evidence uploaded", UPDATED_SSCS, idamTokens);
+    }
+
+    private void addSscsDocumentToCcd(Long ccdCaseId, Document document) {
+        log.info("Adding document to case {} ...",ccdCaseId);
+        IdamTokens idamTokens = idamService.getIdamTokens();
+        log.info("Got idam tokens ");
+
```
Comment (chrisgrimble) [inline src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java:125]: We have already loaded the caseDetails earlier in this call probably needs to be passed through

**Review by chrisgrimble — CHANGES_REQUESTED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java (hunk 2 above)
Comment (chrisdavecm) [reply src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java:17]: Have raised story https://tools.hmcts.net/jira/browse/SSCS-4605 to cover this testing

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/Application.java (hunk 3 above)
Comment (chrisdavecm) [reply src/main/java/uk/gov/hmcts/reform/sscscorbackend/Application.java:60]: We may need this, will address in sprint

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/OnlineHearingPdfWraper.java (hunk 4 above)
Comment (chrisdavecm) [reply src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/OnlineHearingPdfWraper.java:7]: Removed lombok

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/CohEvent.java (hunk 5 above)
Comment (chrisdavecm) [reply src/main/java/uk/gov/hmcts/reform/sscscorbackend/domain/onlinehearing/CohEvent.java:5]: removed

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java (hunk 6 above)
Comment (chrisdavecm) [reply src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java:28]: Need to look into how to best do this

File: src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java (hunk 8 above)
Comment (chrisdavecm) [reply src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/EvidenceUploadService.java:125]: done

File: src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java (hunk 1 above)
Comment (chrisdavecm) [reply src/functionalTest/java/uk/gov/hmcts/reform/sscscorbackend/CreateHearingPdfTest.java:19]: https://tools.hmcts.net/jira/browse/SSCS-4605 should cover this test

**Review by chrisgrimble — APPROVED** [review]

**Pull request merged**